components/social/feed/MainContentSection/index.tsx

```diff
@@ -15,24 +20,43 @@ const MainContentSection = () => {
     reviewData,
   };
 
-  return <MainContentSectionView {...props} />;
+  // infiniteScroll 설명을 위한 임시 작업
+
+  // return <MainContentSectionView {...props} />;
+  return <MainContentSectionView />;
 };
 
 interface MainContentSectionViewProps {
   reviewData?: ReviewResponseType[];
 }
 
-const MainContentSectionView = ({
-  reviewData,
-}: MainContentSectionViewProps) => {
-  return (
-    <S.Section>
-      {reviewData?.map((data: ReviewResponseType, i) => (
-        <Review key={i} data={data} />
-      ))}
-    </S.Section>
-  );
-};
+const MainContentSectionView = () =>
+  //   {
+  //   reviewData,
+  // }: MainContentSectionViewProps
+  {
+    // infiniteScroll 설명을 위한 임시 작업
+
+    const observeTarget = useRef<HTMLDivElement>(null);
+
+    const infiniteQuery = useGetInfiniteFeed();
+
+    useIntersectionObserver({
+      containerTarget: observeTarget,
+      onIntersect: infiniteQuery.fetchNextPage,
+    });
+    const reviewData = linkageInfiniteScrollData<ReviewResponseType>(
+      infiniteQuery?.data
+    );
```

Comment:
`useIntersectionObserver` 내부에서 `useRef`를 사용하도록 변경해서 `useIntersectionObserver`에는 `onIntersect`만 인자로 넘겨주면 되도록 하고 `useIntersectionObserver`의 return 값인 `ref`만 원하는 element에 할당해주면 되는 방식은 어떠신가요??
```suggestion
    const infiniteQuery = useGetInfiniteFeed();
    const { ref } = useIntersectionObserver(infiniteQuery.fetchNextPage);
    // 하나의 배열로 변환하고 싶을 때 사용
    const reviewData = linkageInfiniteScrollData<ReviewResponseType>(
      infiniteQuery?.data
    );
```

Reply:
```
    const infiniteQuery = useGetInfiniteFeed();
    const observeTarget = useIntersectionObserver(infiniteQuery.fetchNextPage);
    const reviewData = linkageInfiniteScrollData<ReviewResponseType>(
      infiniteQuery?.data
    );
```
이렇게 변경했습니다 👍👍 확실히 코드가 더 보기 편해진 것 같아요